### ISSUE TITLE
Data Inspector Segmentation Fault due to Error - "cannot accept empty buffer as protobuf" for empty tiles

### ISSUE THREAD

**Issue by timwaters** [kosmtik/kosmtik#51]
Hello,

I am seeing a seg fault when turning on the Data Inspector when the tile is not showing anything. This is using current master ee1fd3e31f72
```
httpserver] /osm-bright/tile/10/504/334.png?t=1426111602116 200
[httpserver] /osm-bright/tile/10/504/338.png?t=1426111602116 200
[httpserver] /osm-bright/tile/10/508/336.xray?t=1426111672909&layer=&background= 200
[httpserver] /osm-bright/tile/10/508/335.xray?t=1426111672910&layer=&background= 200
[httpserver] /osm-bright/tile/10/507/334.xray?t=1426111672910&layer=&background= 200
[httpserver] /osm-bright/tile/10/507/335.xray?t=1426111672910&layer=&background= 200
[httpserver] /osm-bright/tile/10/507/336.xray?t=1426111672910&layer=&background= 200

/home/tim/projects/kosmtik/src/back/XRayTile.js:18
    vtile.setData(this.data);
          ^
Error: cannot accept empty buffer as protobuf
    at XRayTile.render (/home/tim/projects/kosmtik/src/back/XRayTile.js:18:11)
    at /home/tim/projects/kosmti
```

![image](https://cloud.githubusercontent.com/assets/48941/6607973/190a2fd2-c83b-11e4-82b0-c21826309109.png)

It is able to show the data tiles when there is something rendered on each tile, but appears to happen when there is an empty tile. The background of the tiles are transparent, if that helps, with the project.mml living at https://github.com/timwaters/cloud_mapping/blob/master/osm-bright/osm-bright/project.mml

**Comment by timwaters** [comment]
I have managed to work around this by the following snippet. Would be happy to make a PR for this, but I'm totally unsure if this is the correct solution (I can include the vtile.parse call in the if block and it seems okay also..)

```
XRayTile.prototype.render = function (project, map, cb) {
    var styleMap = this.styleMap(project),
        vtile = new mapnik.VectorTile(this.z, this.x, this.y);

    if (this.data.length > 0){
        vtile.setData(this.data);
    }
    vtile.parse();
    vtile.render(styleMap, new mapnik.Image(256, 256), cb);
};
```

**Comment by yohanboniface** [comment]
Sorry, totally missed your issue (I hate Github notifications) :(

You change sounds good.

About coding conventions, I would remove the `{}` and put that on oneline. That's not the usual Javascript coding style, but my python background need a clean syntax ;)

So something like this:

```
if (this.data.length) vtile.setData(this.data);
```

**Comment by timwaters** [comment]
okay will do, thanks

**Comment by yohanboniface** [comment]
Closed in #53.
